src/ArchiveBuilderHelper.cpp

```diff
@@ -1,66 +1,75 @@
 #include <bsapacker/ArchiveBuilderHelper.h>
 
-#include <QDirIterator>
-#include <QtConcurrent/QtConcurrentMap>
+#include <algorithm>
+#include <boost/algorithm/string.hpp>
 
 #include "SettingsService.h"
 
+using std::filesystem::path;
+using std::filesystem::directory_entry;
+using std::filesystem::directory_iterator;
+using std::filesystem::recursive_directory_iterator;
+
 namespace BsaPacker
 {
-	const std::array <QString, 3> ArchiveBuilderHelper::INCOMPRESSIBLE_TYPES = { "wav", "ogg", "mp3" };
+	const std::array<std::string, 3> ArchiveBuilderHelper::INCOMPRESSIBLE_TYPES = { ".wav", ".ogg", ".mp3" };
```

Comment:
You could use a `std::set` with a custom case-insensitive comparator, this way you could just query `INCOMPRESSIBLE_TYPES.count(ext) > 0` to check.